rcgen/src/key_pair.rs

```diff
@@ -130,7 +132,7 @@ impl KeyPair {
 	pub fn from_pem(pem_str: &str) -> Result<Self, Error> {
 		let private_key = pem::parse(pem_str)._err()?;
 		let private_key_der: &[_] = private_key.contents();
-		Ok(private_key_der.try_into()?)
+		Self::from_der(PrivatePkcs8KeyDer::from(private_key_der))
```

Comment:
Maybe the docstring for this function should document that this function can only consume PKCS#8 keys?